core/Rename.cpp

```diff
@@ -266,30 +267,62 @@ namespace olympia
                 renaming_inst->setStatus(Inst::Status::RENAMED);
                 ILOG("sending inst to dispatch: " << renaming_inst);
 
-                // TODO: Register renaming for sources
                 const auto & srcs = renaming_inst->getSourceOpInfoList();
+                const auto & dests = renaming_inst->getDestOpInfoList();
                 for(const auto & src : srcs)
                 {
-                    const auto rf  = olympia::coreutils::determineRegisterFile(src);
-                    const auto num = src.field_value;
-                    auto & bitmask = renaming_inst->getSrcRegisterBitMask(rf);
-                    const uint32_t prf = map_table_[rf][num];
-                    reference_counter_[rf][prf]++;
-                    renaming_inst->getRenameData().setSource({prf, rf});
-                    bitmask.set(prf);
-                    ILOG("\tsetup source register bit mask "
-                         << sparta::printBitSet(bitmask)
-                         << " for '" << rf << "' scoreboard");
-                }
+                    if(renaming_inst->isLoadStoreInst()){
+                        if(src.field_id == mavis::InstMetaData::OperandFieldID::RS2){
+                            const auto rf  = olympia::coreutils::determineRegisterFile(src);
+                            const auto num = src.field_value;
+                            auto & bitmask = renaming_inst->getDataRegisterBitMask(rf);
+                            const uint32_t prf = map_table_[rf][num];
+                            reference_counter_[rf][prf]++;
+                            renaming_inst->getRenameData().setDataReg({prf, rf});
+                            bitmask.set(prf);
 
-                // TODO: Register renaming for destinations
-                const auto & dests = renaming_inst->getDestOpInfoList();
+                            ILOG("\tsetup store data register bit mask "
+                                << sparta::printBitSet(bitmask)
+                                << " for '" << rf << "' scoreboard");
+                        }
+                        else{
+                            core_types::RegFile rf = core_types::RegFile::RF_INTEGER;
+                            auto & bitmask = renaming_inst->getSrcRegisterBitMask(rf);
+                            const uint32_t prf = freelist_[rf].front();
+                            freelist_[rf].pop();
+                            renaming_inst->getRenameData().setSource({prf, rf});
+                            reference_counter_[rf][prf]++;
```

Comment:
Wait, isn't this a source register?  Why are you grabbing a register from the FL?

Reply:
I was thinking we needed to hold a scoreboard space to allow for address calculation checking to be tracked, so I assumed we needed to allocate a PRF to "hold" that bit. But I realized I went too far ahead in my thinking and created a bug in the process, because I'm always setting that bit to ready and not actually checking if the address is ready. I'll fix/change it back to only source renaming checking/mapping.

Just to make sure we're on the same page, the goal of the address/data separation is only meant to create separate data structures for tracking either address or data, and then Suraj and the team will handle adding in a way to track the address calculation if it runs ahead?

Reply:
That's correct.  The address is *only* calculated in the LS and usually store alongside the operation in either a load queue or a store queue.  It will *never* write that value to a PRF -- no reason for that.

Reply:
Fixed in [5212a4c](https://github.com/riscv-software-src/riscv-perf-model/pull/70/commits/5212a4c3274e758abca7bebf8ebc45a4ac210ca9)